src/FDC3NotificationService/index.tsx

```diff
@@ -0,0 +1,251 @@
+			metadata: {
+				source: source?.metadata?.source,
+				timeout: notification.timeout,
+				isDeleted: notification.isDeleted,
+				isMuted: notification.isMuted,
+				isRead: notification.isRead,
+				isSnoozed: notification.isSnoozed,
+				issuedAt: notification.issuedAt,
+				receivedAt: notification.receivedAt,
+			},
+		};
+	};
+
+	const createNotificationListener = (context: FDC3Notification): Promise<Context> => {
+		const { type } = context;
+		return new Promise(async (resolve) => {
+			if (type === "fdc3.notification") {
+				const notification: INotification = convertFDC3NotificationToFinsemble(context);
+
+				/**
+				 * Subscribe to the notification returned from the notification service
+				 */
+				const subscription = await FSBL.Clients.NotificationClient.subscribe(
+					{
+						filter: { include: [{ id: notification.id as string }] },
+					},
+					(processedNotification) => {
+						FSBL.Clients.NotificationClient.unsubscribe(subscription.id);
+						const returnContext = convertFinsembleToFDC3(processedNotification);
+						resolve(returnContext);
+					}
+				);
+
+				FSBL.Clients.NotificationClient.notify(notification);
+			}
+		});
+	};
+
+	const getNotificationsListener = (context: FDC3NotificationFilter): Promise<Channel> =>
+		new Promise<Channel>(async (resolve, reject) => {
+			let filter: Partial<ISubscription> = {
+				filter: convertFDC3FilterToFinsemble(context),
+			};
+
+			const channel = await fdc3.createPrivateChannel();
+			const subscription = await FSBL.Clients.NotificationClient.subscribe(filter, (notification) => {
+				channel.broadcast(convertFinsembleToFDC3(notification));
+			});
+
+			channel.onDisconnect(() => {
+				// Not getting called when the window closes - can end up with a lot of unnecessary router traffic
+				FSBL.Clients.NotificationClient.unsubscribe(subscription.id);
+			});
+			resolve(channel);
+			reject(ChannelError.CreationFailed);
+		});
+
+	const registerIntents = () => {
+		FSBL.Clients.Logger.log("Adding intent listeners");
+		fdc3.addIntentListener("CreateNotification", createNotificationListener as IntentHandler);
+		fdc3.addIntentListener("UpdateNotification", createNotificationListener as IntentHandler);
+		fdc3.addIntentListener("GetNotifications", getNotificationsListener);
+	};
+
+	const performAction: StandardErrorCallback<RouterResponse<CustomActionResponse<FDC3Action>> | null> = (
+		err,
+		response
+	) => {
+		if (response?.data.actionPayload.type == "fdc3.action" && response?.data.actionPayload.context) {
+			const { actionPayload: action } = response.data;
+			if (action.intent) {
+				fdc3.raiseIntent(action.intent, action.context, action.app);
+			} else {
+				fdc3.raiseIntentForContext(action.context);
+			}
+		}
+	};
+
+	const createRouterEndpoints = () => {
+		FSBL.Clients.RouterClient.addListener(FDC3_CUSTOM_ACTION_CHANNEL, performAction);
+	};
+
+	const initService = () => {
+		registerIntents();
+		createRouterEndpoints();
+		FSBL.publishReady();
```

Comment:
Looks like there was a mistake in the config. You removed `waitForInitiailzation: false`, but it looks like it should have been set to true. Sorry I didn't catch this before.

Reply:
Ahh yeah. This was false because of a bug around the time of the original FDC3 notifications PoC. I think it's been fixed. Will check.

Reply:
https://cosaic.kanbanize.com/ctrl_board/106/cards/49865/details/